include/openthread/platform/border_routing.h

```diff
@@ -65,6 +65,18 @@ extern "C" {
  */
 extern void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
 
+/**
+ * Used to notify the platforms whether they should request an prefix via DHCPv6 PD for Thread interface.
+ * When aShouldRequest is false, the platform must release the prefix lease.
+ *
+ * Requires `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE`.
+ *
+ * @param[in] aInstance  A pointer to an OpenThread instance.
+ * @param[in] aRequest  Indicate if the platform need to request/release the prefix.
+ *
+ */
+void otPlatBorderRoutingHandleDhcp6PdStateChange(otInstance *aInstance, bool aRequest);
```

Comment:
If you use ...StateChange as the name of the API, I would prefer passing the state enum instead of a boolean to the platform.

Reply:
Also, I would suggest update the statement to include that:

- WHen the state is Running, the platform should request a prefix via DHCPv6 PD
- For other states, the platform should not request a prefix and release the DHCPv6 PD lease.